pyQuARC/code/custom_validator.py

```diff
@@ -102,3 +95,106 @@ def bounding_coordinate_logic_check(coordinates_dictionary):
         )
 
         return result
+
+    @staticmethod
+    def presence_check(*field_values):
+        """
+            Checks if one of the field has a value
+        """
+        # At least one of all the fields should have a value
+        # It is basically a OneOf check
+        validity = False
+        value = None
+
+        for field_value in field_values:
+            if field_value:
+                value = field_value
+                validity = True
+
+        return {
+            "valid": validity,
+            "value": value
+        }
+
+    @staticmethod
+    @if_arg
+    def opendap_url_location_check(field_value):
+        # The field shouldn't have a opendap url
+        return {
+            "valid": 'opendap' not in field_value.lower(),
+            "value": field_value
+        }
+
+    @staticmethod
+    @if_arg
+    def user_services_check(first_name, middle_name, last_name):
+        return {
+            "valid": not (first_name.lower() == 'user' and last_name.lower() == 'services' and (not middle_name)),
+            "value": f'{first_name} {last_name}'
+        }
+
+    @staticmethod
+    def doi_missing_reason_explanation(explanation, missing_reason, doi):
+        return {
+            "valid": not((not doi) and (missing_reason) and (not explanation)),
+            "value": explanation
+        }
+
+    @staticmethod
+    @if_arg
+    def boolean_check(field_value):
+        # Checks if the value is a boolean, basically 'true' or 'false' or their case variants
+        return {
+            "valid": field_value.lower() in ["true", "false"],
+            "value": field_value
+        }
+
+    @staticmethod
+    @if_arg
+    def collection_progress_consistency_check(collection_state, ends_at_present_flag, ending_date_time):
+        # Logic: https://github.com/NASA-IMPACT/pyQuARC/issues/61
+        validity = True
+        if collection_state.upper() in ["ACTIVE", "IN WORK"]:
+            validity = (not bool(ending_date_time)) and ends_at_present_flag.lower() == "true"
+        elif collection_state.upper() == "COMPLETE":
+            validity = bool(ending_date_time) and (not bool(ends_at_present_flag) or ends_at_present_flag.lower() == "false")
+        else:
+            validity = False
+        return {
+            "valid": validity,
+            "value": collection_state
+        }
+
+    @staticmethod
+    @if_arg
+    def characteristic_name_uniqueness_check(characteristics):
+        seen, duplicates = set(), set()
+        for characteristic in characteristics['Characteristic']:
+            name = characteristic['Name']
+            if name in seen:
+                duplicates.add(name)
+            else:
+                seen.add(name)
+
+        return {
+            "valid": not bool(duplicates),
+            "value": ', '.join(duplicates)
+        }
+
+    @staticmethod
+    def get_data_url_check(metadata_json):
```

Comment:
Change this to local constant

Reply:
https://github.com/NASA-IMPACT/pyQuARC/pull/84/commits/312bad0b936cf55f8328625e4052617ffd41c46f